pkg/controlsvc/reload.go

```diff
@@ -16,34 +17,29 @@ type (
 	reloadCommand     struct{}
 )
 
-var configPath = ""
+var (
+	configPath       = ""
+	mu               sync.Mutex
```

Comment:
A global mutex is allowed but I would suggest to bundle it up with all the other vars here and create a struct to allow grouping and encapsulation.